metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuiteDiscoveryResult.scala

```diff
@@ -0,0 +1,45 @@
+package scala.meta.internal.metals.testProvider
+
+import javax.annotation.Nullable
+
+import scala.meta.internal.metals.MetalsEnrichments._
+
+import org.eclipse.{lsp4j => l}
+
+final case class TestSuiteDiscoveryResult(
+    targetName: String,
+    targetUri: String,
+    discovered: java.util.List[TestSuiteDiscoveryResult.Discovered]
+)
+
+object TestSuiteDiscoveryResult {
+  sealed trait Discovered {
+    def nonEmpty: Boolean = this match {
+      case p: Package => p.children.asScala.nonEmpty
+      case _: TestSuite => true
+    }
+  }
+
+  final class Package private (
```

Comment:
Why do we need package? Can't we have just a list of TestSuite ? This would leave us with just one class instead of hierarchy.

Reply:
Could you provide an example? I think it's doable to get rid of hierarchy, but I don't see how to model following with just List, without Package at all.
```
TestSuite1
a.TestSuite2
a.b.TestSuite3
```

Reply:
But then the client will have to map this flat structure into the hierarchy itself. I assumed that it's easier to do this once on the server-side, because the client can always flat this back into the list and grouping is slightly harder.

To sum up, I don't mind moving this logic to the client side.

Reply:
@ckipp01 @ayoub-benali How this looks like from the other editors' side? What would be a better format for nvim or sublime to process?

Reply:
Code shouldn't be vscode specific and should be easy to reuse by other editors as well.